hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/UuidKeyGenerator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.keygen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
+
+/**
+ * A KeyGenerator which use the uuid as the record key.
+ */
+public class UuidKeyGenerator extends BuiltinKeyGenerator {
```

Comment:
This is used for `InsertInto` for hudi table without a primary key. Currently, we must return a `RecordKey` for each record when write to hudi. If there user has not specify a primary key in the `DDL`. I provide a default `UuidKeyGenerator` which generate uuid for the record.

Reply:
Could we make this key be prefixed by commit time? this will result in ordered keys, which will improve range/index lookup for update/deletes down the line. Random UUID ranges will always overlap each other, so given a key `K` , it's very likely that the this is true for almost all files written like this => `minKey < K < maxKey`.


Actually, `_hoodie_commit_seqno` would be a good key for example for insert only datasets. Can we generate something like that instead of uuid, which has poor ordering.